src/App.jsx

```diff
@@ -13,34 +17,76 @@
   'Jam',
   'Garlic',
 ];
+export const App = () => {
+  const [sortType, setSortType] = useState(SORT_BY_FIELDS.none);
+  const [isReversed, setIsReversed] = useState(false);
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+  const resetGoods = () => {
+    setSortType(SORT_BY_FIELDS.none);
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const visibleGoods = getVisibleGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+    SORT_BY_FIELDS,
+  );
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortType(SORT_BY_FIELDS.alphabetic)}
+          type="button"
+          className={cn('button', {
+            'is-info': sortType === SORT_BY_FIELDS.alphabetic,
+            'is-light': sortType !== SORT_BY_FIELDS.alphabetic,
+          })}
+        >
+          Sort alphabetically
+        </button>
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        <button
+          onClick={() => setSortType(SORT_BY_FIELDS.length)}
+          type="button"
+          className={cn('button', {
+            'is-success': sortType === SORT_BY_FIELDS.length,
+            'is-light': sortType !== SORT_BY_FIELDS.length,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={cn('button', {
+            'is-warning': isReversed,
+            'is-light': !isReversed,
+          })}
+        >
+          Reverse
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        {(sortType !== SORT_BY_FIELDS.none || isReversed) && (
+          <button
+            onClick={() => resetGoods()}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
```

Comment:
The `Reset` button should only be visible if the goods are not already in the original order. Ensure the visibility logic for this button adheres to this requirement.